libs/wai-utilities/src/Network/Wai/Utilities/Error.hs

```diff
@@ -18,40 +18,78 @@
 -- You should have received a copy of the GNU Affero General Public License along
 -- with this program. If not, see <https://www.gnu.org/licenses/>.
 
-module Network.Wai.Utilities.Error where
+module Network.Wai.Utilities.Error
+  ( Error (..),
+    ErrorData (..),
+    mkError,
+    (!>>),
+    byteStringError,
+  )
+where
 
 import Control.Error
 import Data.Aeson hiding (Error)
+import Data.Aeson.Types (Pair)
+import Data.Domain
 import Data.Text.Lazy.Encoding (decodeUtf8)
 import Imports
 import Network.HTTP.Types
 
 data Error = Error
   { code :: !Status,
     label :: !LText,
-    message :: !LText
+    message :: !LText,
+    errorData :: Maybe ErrorData
   }
   deriving (Show, Typeable)
 
+mkError :: Status -> LText -> LText -> Error
+mkError c l m = Error c l m Nothing
+
 instance Exception Error
 
+data ErrorData = FederationErrorData
+  { federrDomain :: !Domain,
```

Comment:
Is this domain the originating domain or the remote domain? Possibly `FederationErrorData` could be extended in the future to contain both.

Reply:
It's the remote domain. I assumed the originating domain would not be needed because it would simply be the one where the instance is running. But yes, it can be added easily.